src/classes/RD2_NamingService.cls

```diff
@@ -203,51 +211,250 @@ public with sharing class RD2_NamingService {
     * @param donorById Account/Contact donor record by Id
     * @return String New name for the Recurring Donation record
     */
-    private String constructName(npe03__Recurring_Donation__c rd,  Map<Id, RD2_Donor.Record> donorById) {
-        Boolean useContact = rd.npe03__Organization__c == null;
-        if (!useContact) {
-            useContact = donorById.get(rd.npe03__Organization__c).recordTypeId == hhRecordTypeId;
+    private String constructName(npe03__Recurring_Donation__c rd, Map<Id, RD2_Donor.Record> donorById) {
+
+        NameBuilder nameBuilder = new NameBuilder(rd, new NameFormatter())
```

Comment:
The name formatter should be initialized once as the `RD2_NamingService` var and used to construct names many times.
```

    private NameFormatter nameFormatter {
        get {
            if (nameFormatter == null) {
                nameFormatter = new NameFormatter();
            }
            return nameFormatter;
        }
        set;
    }
```

Reply:
Thanks @RadSFDeveloper. That was a key part of the previous refactor that I just completely overlooked. Changes have been made.